src/components/dashboard/menu-table/popup-forms.js

```diff
@@ -255,26 +350,35 @@ const EditDishForm = (props) => {
             //show some error
         }
     }
-
     const updateCategorySelection = (category) => {
         console.log("category selection updated")
         console.log(category)
-        setCategoryId(category.id)
+        setCategoryId(category)
     }
 
     return (
         <>
-            <ModalBackground/>
+            <ModalBackground />
             <StyledModal>
                 <Container>
-                    <FormInput placeholder="Dish Name" value={name} name='name' onChange={ (event) => {setName(event.target.value)} }/>
-                    <Dropdown categoryId={categoryId} updateSelection={updateCategorySelection} menuId={props.menuId}/>
-                    <FormInput placeholder="Description" value={ description } name='description' onChange={ (event) => {setDescription(event.target.value)} }/>
-                    <FormInput placeholder="Price" name='price' onChange={ (event) => {setPrice(event.target.value)} }/>
-                    <TagsForm tags={ dishTags } setTags={ setDishTags }></TagsForm>
+                    <DishFormTitle>Update Dish</DishFormTitle>
+                    <DishFormSubtitle>Dish Name</DishFormSubtitle>
+                    <DishFormInput placeholder="Change dish name..." value={name} name='name' onChange={(event) => { setName(event.target.value) }} />
+                    <Divider color="#DCE2E9" />
+                    <DishFormSubtitle>Menu Category</DishFormSubtitle>
+                    <CategoryDropdown categoryId={categoryId} updateSelection={updateCategorySelection} menuId={props.menuId} />
+                    <Divider color="#DCE2E9" />
+                    <DishFormSubtitle>Description</DishFormSubtitle>
+                    <DishFormInput placeholder="Change description..." value={description} name='description' onChange={(event) => { setDescription(event.target.value) }} />
+                    <Divider color="#DCE2E9" />
+                    <DishFormSubtitle>Price</DishFormSubtitle>
+                    <DishFormInput placeholder="Change price..." name='price' onChange={(event) => { setPrice(event.target.value) }} />
+                    <Divider color="#DCE2E9" />
+                    <DishFormSubtitle>Allergen Search</DishFormSubtitle>
+                    <TagsForm tags={dishTags} setTags={setDishTags} existTags={props.dish.Tags}></TagsForm>
```

Comment:
From line 313, existTags is the same as dishTags. That can help you simplify the logic when you fill in defaultTags in the drop down I believe. e.g. you don't need to check whether existTag is undefined. Just check if tags is empty (when creating it starts out empty and when editing it has some tags in it)